src/build/arg_group.rs

```diff
@@ -545,9 +543,24 @@ mod test {
             .requires_all(&["r2", "r3"])
             .requires("r4");
 
-        let args = vec!["a1".key(), "a4".key(), "a2".key(), "a3".key()];
-        let reqs = vec!["r1".key(), "r2".key(), "r3".key(), "r4".key()];
-        let confs = vec!["c1".key(), "c2".key(), "c3".key(), "c4".key()];
+        let args = vec![
+            Id::from("a1"),
+            Id::from("a4"),
```

Comment:
Because, as you can see a dozen lines below, the `args` array is only used in the assert. Dear `rustc` can figure that the elements of the array must implement `Debug & PartialEq`, but it doesn't anything about the concrete type, so type inference fails. 

I had to specify the type explicitly to overcome that.